src/AppBundle/Controller/DeleteCounterController.php

```diff
@@ -0,0 +1,198 @@
+use Symfony\Component\HttpFoundation\Response;
+use Xtools\Project;
+use Xtools\ProjectRepository;
+use Xtools\User;
+use Xtools\UserRepository;
+
+/**
+ * This controller handles the Simple Edit Counter tool.
+ */
+class DeleteCounterController extends Controller
+{
+
+    /**
+     * Get the tool's shortname.
+     *
+     * @return string
+     */
+    public function getToolShortname()
+    {
+        return 'dc';
+    }
+
+    /**
+     * The Simple Edit Counter search form.
+     *
+     * @param Request $request The HTTP request.
+     * @param string  $project The project database name or domain.
+     *
+     * @return Response
+     *
+     * @Route("/dc",           name="dc")
+     * @Route("/dc",           name="DeletionCounter")
+     * @Route("/dc/",          name="DeletionCounterSlash")
+     * @Route("/dc/index.php", name="DeletionCounterIndexPhp")
+     * @Route("/dc/{project}", name="DeletionCounterProject")
+     */
+    public function indexAction(Request $request, $project = null)
+    {
+        // Get the query parameters.
+        $projectName = $project ?: $request->query->get('project');
+        $username = $request->query->get(
+            'username',
+            $request->query->get('user')
+        );
+
+        // If we've got a project and user, redirect to results.
+        if ($projectName != '' && $username != '') {
+            $routeParams = [ 'project' => $projectName, 'username' => $username ];
+            return $this->redirectToRoute(
+                'DeletionCounterResult',
+                $routeParams
+            );
+        }
+
+        // Instantiate the project if we can, or use the default.
+        $theProject = (!empty($projectName))
+            ? ProjectRepository::getProject($projectName, $this->container)
+            : ProjectRepository::getDefaultProject($this->container);
+
+        // Show the form.
+        return $this->render(
+            'deletionCounter/index.html.twig',
+            [
+            'xtPageTitle' => 'tool-dc',
+            'xtSubtitle' => 'tool-dc-desc',
+            'xtPage' => 'dc',
+            'project' => $theProject,
+            ]
+        );
+    }
+
+    /**
+     * Display the results
+     *
+     * @param string $project  The project domain name.
+     * @param string $username The username.
+     *
+     * @return Response
+     *
+     * @Route("/dc/{project}/{username}", name="DeletionCounterResult")
+     */
+    public function resultAction($project, $username)
+    {
+        /**
+         * Project information
+         *
+         * @var Project $project
+         */
+        $project = ProjectRepository::getProject($project, $this->container);
+        $projectRepo = $project->getRepository();
+
+        $user = UserRepository::getUser($username, $this->container);
+
+        $userID = $user->getId($project);
+
+        if (!$project->exists()) {
+            $this->addFlash('notice', ['invalid-project', $project]);
```

Comment:
This one seems to have merit: `$project` at this point is a `Project` and not a string as `addFlash` expects